make/libraries

```diff
@@ -166,7 +166,7 @@ endif
 $(TBB_BIN)/tbb.def: $(TBB_BIN)/tbb-make-check $(TBB_BIN)/tbbmalloc.def
 	@mkdir -p $(TBB_BIN)
 	touch $(TBB_BIN)/version_$(notdir $(TBB))
-	tbb_root="$(TBB_RELATIVE_PATH)" CXX="$(CXX)" CC="$(TBB_CC)" LDFLAGS='$(LDFLAGS_TBB)' '$(MAKE)' -C "$(TBB_BIN)" -r -f "$(TBB_ABSOLUTE_PATH)/build/Makefile.tbb" compiler=$(TBB_CXX_TYPE) cfg=release stdver=c++1y  CXXFLAGS="$(TBB_CXXFLAGS)"
+	tbb_root="$(TBB_RELATIVE_PATH)" CXX="$(CXX)" CC="$(TBB_CC)" LDFLAGS='$(LDFLAGS_TBB)' '$(MAKE)' -C "$(TBB_BIN)" -r -f "$(TBB_ABSOLUTE_PATH)/build/Makefile.tbb" compiler=$(TBB_CXX_TYPE) cfg=release stdver=c++1y  CXXFLAGS="$(TBB_CXXFLAGS) -Wno-unused-but-set-variable"
```

Comment:
Wouldn't it be better to add this flag in line 125 of the file to the clang specific options set as default?

Reply:
much better - thanks!

Reply:
@mitzimorris This isn't where I thought the change would be. I thought we would have applied it to just the dylib target that you mentioned. Do we have a separate dylib target?

I didn't realize the change is clang-specific. Does g++ error out with this option defined?

Reply:
@syclik - regarding the change - this defines the way that several targets, including libtbb.dylib, line 175, are invoked.

I'm not sure if the build warnings are specific to clang - I believe the `sprintf` warnings are, and that one is particularly vexing.  however, there's a better and more general solution:

instead of including 3rd party libs using "-I", use compiler directive "-isystem" - cf https://stackoverflow.com/a/2452392

changing line 108 of `make/compiler_flags` from 
```
INC ?= $(INC_FIRST) -I $(if $(MATH),$(MATH),.) -I $(EIGEN) -I $(BOOST) $(INC_SUNDIALS)
```
to 
```
INC ?= $(INC_FIRST) -I $(if $(MATH),$(MATH),.) -I $(EIGEN) -isystem $(BOOST) $(INC_SUNDIALS)
```

gets rid of all the `sprintf` spew.